File: Testing/Code for Testing/Early Basic Syntax Testing/pipe/test_pipe.js

function twice(x) {
    return (x * 2);
}

function add_one(x) {
    return (x + 1);
}

function flip_sign(x) {
    return (0 - x);
}

function main() {
    let a = twice(5);
    console.log(a);
    let b = add_one(twice(5));
    console.log(b);
    let c = flip_sign(add_one(twice(3)));
    console.log(c);
    let d = add_one(10);
    console.log(d);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Code for Testing/Early Basic Syntax Testing/pipe/test_pipe.py
from dataclasses import dataclass, field

def _ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class _ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def twice(x):
    return (x * 2)

def add_one(x):
    return (x + 1)

def flip_sign(x):
    return (0 - x)

def main():
    a = twice(5)
    print(a)
    b = add_one(twice(5))
    print(b)
    c = flip_sign(add_one(twice(3)))
    print(c)
    d = add_one(10)
    print(d)
    return 0


if __name__ == "__main__":
    main()
